src/graph/YieldExecutor.cpp

```diff
@@ -266,32 +369,67 @@ Status YieldExecutor::executeConstant() {
     std::vector<VariantType> values;
     values.reserve(size);
 
-    for (auto *col : yields_) {
-        auto expr = col->expr();
-        auto v = expr->eval();
-        if (!v.ok()) {
-            return v.status();
+    if (aggFuns_.empty()) {
+        for (auto *col : yields_) {
+            auto expr = col->expr();
+            auto v = expr->eval();
+            if (!v.ok()) {
+                return v.status();
+            }
+            values.emplace_back(v.value());
         }
-        values.emplace_back(v.value());
-    }
 
-    auto outputSchema = std::make_shared<SchemaWriter>();
-    auto status = Collector::getSchema(
-            values, resultColNames_, colTypes_, outputSchema.get());
-    if (!status.ok()) {
-        return status;
+        auto outputSchema = std::make_shared<SchemaWriter>();
+        auto status = Collector::getSchema(
+                values, resultColNames_, colTypes_, outputSchema.get());
+        if (!status.ok()) {
+            return status;
+        }
+
+        RowWriter writer(outputSchema);
+        auto rsWriter = std::make_unique<RowSetWriter>(outputSchema);
+        for (auto col : values) {
+            status = Collector::collect(col, &writer);
+            if (!status.ok()) {
+                return status;
+            }
+        }
+        rsWriter->addRow(writer);
+        finishExecution(std::move(rsWriter));
+    } else {
+        auto i = 0u;
+        for (auto col : yields_) {
+            auto *expr = col->expr();
+            auto value = expr->eval();
+            if (!value.ok()) {
+                return value.status();
+            }
+            auto val = toColumnValue(value.value(), cpp2::ColumnValue::Type::__EMPTY__);
+            aggFuns_[i]->apply(val);
+            i++;
+        }
     }
 
-    RowWriter writer(outputSchema);
-    auto rsWriter = std::make_unique<RowSetWriter>(outputSchema);
-    for (auto col : values) {
-        status = Collector::collect(col, &writer);
+    if (!aggFuns_.empty()) {
+        cpp2::RowValue row;
+        auto schema = std::make_shared<SchemaWriter>();
+        std::vector<cpp2::ColumnValue> cols;
+        auto i = 0u;
+        for (auto f : aggFuns_) {
+            auto val = f->getResult();
+            schema->appendCol(resultColNames_[i],
+                                static_cast<nebula::cpp2::SupportedType>(val.getType()));
+            cols.emplace_back(std::move(val));
+            i++;
+        }
+        row.set_columns(std::move(cols));
+        auto rsWriter = std::make_unique<RowSetWriter>(schema);
+        auto status = getAggResultWriter(std::move(row), rsWriter.get());
         if (!status.ok()) {
             return status;
         }
+        finishExecution(std::move(rsWriter));
```

Comment:
It seems we can put them into single block. And it's better to make it as a function.